drivers/flash/soc_flash_b9x.c

```diff
@@ -15,6 +15,7 @@
 #include <zephyr/device.h>
 #include <zephyr/drivers/flash.h>
 #include <zephyr/kernel.h>
+#include <stdlib.h>
```

Comment:
The driver is already using k_malloc which allocates data in system heap not user.
For now it's using both heaps.
Clear way is to use only system heap for drivers. But if we want to have memory pofit lets use user in all driver code, anyway we have no MPU existed/setuped.

Reply:
the init config heap is configed to 512, but the heap of the lib is very large(12k) , so I think there is no need to enlarge the system heap , multi use of the heap .